src/worldstate/src/routines/BottomLine/FindLine.cpp

```diff
@@ -0,0 +1,46 @@
+//
+// Created by drdc-s2632 on 17/07/18.
+//
+
+#include "FindLine.h"
+#include <worldstate/StateMsg.h>
+
+std::vector<ros::Subscriber>
+FindLine::getNodeSubscriptions(ros::NodeHandle nh) {
+    std::string line_detect_topic = "line_detect_output";
+
+    timer_ =
+    nh.createTimer(ros::Duration(30), &FindLine::timerCallback, this, false);
+    timer_.start();
+
+    std::vector<ros::Subscriber> subs;
+
+    subs.push_back(
+    nh.subscribe(line_detect_topic, 10, &FindLine::lineDetectCallback, this));
+
+    return subs;
+}
+
+void FindLine::lineDetectCallback(
+const line_detect::LineDetectMsg::ConstPtr& msg) {
+    worldstate::StateMsg stateMsg;
+
+    stateMsg.state = worldstate::StateMsg::findingLine;
+
+    if (!(msg->lateralDistanceFromFrontMarker == -1.0f ||
+          msg->distanceFromEndOfFrontMarker == -1.0f ||
+          msg->angleToParallelFrontMarker == -1.0f)) {
+        timer_.stop();
+        stateMsg.state = worldstate::StateMsg::adjustingToLine;
+    }
+
+    publishNextState(stateMsg);
+}
+
+void FindLine::timerCallback(const ros::TimerEvent& event) {
+    worldstate::StateMsg stateMsg;
+
+    // stateMsg.state = worldstate::StateMsg::locatingDie;
```

Comment:
should these be commented out? doing nothing here.

Reply:
Since we don't have a subroutine set up named locatingDie these calls make the worldstatenode crash

Reply:
ahhh, fair